variables.tf

```
@@ -0,0 +1,103 @@
+variable "namespace" {
+  type = string
+  default = "kube-prometheus-stack"
+}
+
+variable "chart_version" {
+  type = string
+}
+
+variable "grafana_pv_size" {
+  type = string
+  default = "10Gi"
+}
+
+variable "prometheus_pv_size" {
+  type = string
+  default = "100Gi"
+}
+
+variable "ingress_host" {
+  type = string
+}
+
+variable "basic_auth_username" {
+  type = string
+  default = "admin"
+}
+
+variable "cluster_issuer_name" {
+  type = string
+}
+
+variable "install_adapter" {
+  type = bool
+  default = false
+}
+
+variable "grafana_ingress_enabled" {
+  type = bool
+  default = true
+}
+
+variable "adapter_chart_version" {
+  type = string
+  default = ""
+}
+
+variable "pull_secrets" {
+  type = bool
+}
+
+variable "regcred" {
+  type = string
+}
+
+variable "secret_name" {
+  type = string
+}
+
+variable "storage_class_name" {
+  type = string
+  default = ""
+}
+
+variable "prometheus_resources" {
+  type        = any
```

I thinkt that here we can define a specific object, like:

```suggestion
  type        = object({ cpu_requests = "",  memory_requests ="" })
```

What do you think?

---

I added this:
```
default     = { 
                  cpu_requests = "", 
                  memory_requests = "" 
                }
```
what do you think?

---

This is ok for the default value, but we can also force the variable to be an object with these two fields.